core/Piranha.AspNetCore/Models/SinglePage.cs

```diff
@@ -53,7 +53,7 @@ public SinglePage(IApi api, IModelLoader loader)
         /// </summary>
         /// <param name="id">The requested model id</param>
         /// <param name="draft">If the draft should be fetched</param>
-        public virtual async Task<IActionResult> OnGet(Guid id, bool draft = false)
+        public virtual async Task<IActionResult> OnGet(Guid id, string pageid = null, bool draft = false, bool preview = false)
```

Comment:
I don't understand what this is supposed to to. In the original version `id` **is** the Page Id, so what is the meaning of the parameter `pageid`. The existing parameter `draft` is used when the request is a preview request, so what is then the meaning of the parameter `preview`?

Reply:
@tidyui  Those parameters are needed in the query string for the middleware to know if we want to preview a specific page, regardless of which site we are currently signed into.